packages/nimble-components/src/rich-text/editor/index.ts

```diff
@@ -335,6 +336,45 @@ export class RichTextEditor extends FoundationElement implements ErrorPattern {
 
     private createTiptapEditor(): Editor {
         const customLink = this.getCustomLinkExtension();
+        const validAbsoluteLinkRegex = /^https?:\/\//i;
+
+        /**
+         * @param slice contains the Fragment of the copied content. If the content is a link, the slice contains anchor text node with link mark.
+         * ProseMirror reference for `transformPasted`: https://prosemirror.net/docs/ref/#view.EditorProps.transformPasted
+         */
+        const transformPasted = (slice: Slice): Slice => {
+            const updateLinkNodes = (nodeArray: NodeObj[]): void => {
+                nodeArray.forEach((node: NodeObj) => {
+                    if (node.type !== 'text' && node.content) {
+                        updateLinkNodes(node.content);
+                    } else {
+                        node.marks?.forEach((mark, index: number) => {
+                            if (mark.type === 'link' && mark.attrs) {
+                                if (
+                                    validAbsoluteLinkRegex.test(mark.attrs.href)
+                                ) {
+                                    node.text = mark.attrs.href;
+                                } else {
+                                    node.marks!.splice(index, 1);
+                                }
+                            }
+                        });
+                    }
+                });
+            };
+            const fragmentAsJson = (slice.content.toJSON() as NodeObj[]) ?? [];
+            updateLinkNodes(fragmentAsJson);
+            const modifiedFragment = Fragment.fromJSON(
+                this.tiptapEditor.schema,
+                fragmentAsJson
+            );
```

Comment:
Initially, we tried using the prose-mirror model to manipulate the fragment, but the [text content](https://prosemirror.net/docs/ref/#model.Node.text) of any node is a read-only property. 

The other approach we also tried is, after finding the `link` mark in a `text` node, thinking of creating a new `text` node with the same marks and replacing it in the same object. However, we felt that this operation involves modifying the node replacements which might cause an issue with the other supported formatings. We also couldn't find a way to just update the text content without affecting the existing nodes. So we decided to utilize the JSON formats to update the text nodes without altering the existing nodes.

Reply:
If this workaround is not temporary it seems like it would be better to understand the prose-mirror model behavior better. I would be surprised if the model cannot handle updating the text for a node.

If this is all a temporary workaround that will not need to be maintained long-term across prose mirror updates that will be scheduled to removed soon then the weaker typing being done here may be a an ok temporary implementation.

Reply:
After exploring deep into the ProseMirror model, found that we can utilize the given APIs to update the Link node value. I have updated the logic without converting to JSON formats and retains all the type information that ProseMirror offers.

We also tested all different kinds of content copied from external sources and verified it worked as expected. So we believe this can be more cleaner approach than the previous one. This method can be updated/removed once we support hyperlinks. I have added the necessary code comments too.